prov/efa/src/rdm/rxr_ep.h

```diff
@@ -279,6 +285,48 @@ static inline int rxr_need_sas_ordering(struct rxr_ep *ep)
 	return ep->msg_order & FI_ORDER_SAS;
 }
 
+static inline uint64_t rxr_get_host_id(char *host_id_file)
+{
+	FILE *fp = NULL;
+	const size_t HOST_ID_LENGTH = 16;
+	char host_id_str[HOST_ID_LENGTH];
```

Comment:
Need a +1 for trailing \0 so strtoul doesn't overrun

Reply:
Thanks for being insistent. I just realized that this is a bug and undefined behavior. So undefined that all my unit test cases passed.